File: frameworks/go/gin/Implementation/compressed.go
package implementation

import (
	"github.com/gin-contrib/gzip"
	"github.com/gin-gonic/gin"
)

// compressedRoutes answer like any other. gin-contrib/gzip, on the /compressed group alone,
// gzips the answer when the request asks for it.
func compressedRoutes(r *gin.Engine, p *Payloads) {
	// rb:wiring compressed.*
	compressed := r.Group("/compressed", gzip.Gzip(gzip.DefaultCompression))

	// rb:handler compressed.gzip_small,compressed.identity_small
	compressed.GET("/small", fresh(&p.Small))

	// rb:handler compressed.gzip_large,compressed.identity_large
	compressed.GET("/large", fresh(&p.Large))
}

File: frameworks/rust/actix-web/Implementation/routes/compressed.rs
use actix_web::HttpResponse;
use actix_web::middleware::Compress;
use actix_web::web::{self, ServiceConfig};

use crate::Payloads;
use crate::serial;

/// compressed: these routes answer like any other. actix-web's Compress middleware, wrapped around
/// this family's scope alone, gzips the answer when the request asks for it.
pub fn configure(cfg: &mut ServiceConfig, p: &'static Payloads) {
    cfg.service(
        web::scope("/compressed")
            // rb:wiring compressed.*
            // Compress takes no level. actix-http gzips at flate2's fast level, which is level 1,
            // and compresses a body of any size.
            .wrap(Compress::default())
            // rb:handler compressed.gzip_small,compressed.identity_small
            .route("/small", web::get().to(move || async move { HttpResponse::Ok().insert_header(serial::fresh()).json(&p.small) }))
            // rb:handler compressed.gzip_large,compressed.identity_large
            .route("/large", web::get().to(move || async move { HttpResponse::Ok().insert_header(serial::fresh()).json(&p.large) })),
    );
}
